src/digest.c

```diff
@@ -37,12 +37,45 @@
 #include "xxhash.h"
 #include "pmurhash.h"
 
+#ifdef _WIN32
+#include <Windows.h>
+#endif
+
 unsigned long ZEXPORT digest_crc32(unsigned long crc,
                                    const unsigned char FAR *buf,
                                    unsigned len);
 
 static const char *sha2_hex_digits = "0123456789abcdef";
 
+#ifdef _WIN32
+FILE* my_open(const char* txt) {
+    FILE* out;
+    wchar_t* buf;
+    size_t len = MultiByteToWideChar(CP_UTF8, 0, txt, -1, NULL, 0);
+    if (len <= 0) {
+        error("Cannot convert file to Unicode: %s", txt);
+    }
+    buf = (wchar_t*) R_alloc(len, sizeof(wchar_t));
+    if (buf == NULL) {
+        error("Could not allocate buffer of size: %ll", len);
+    }
+
+    MultiByteToWideChar(CP_UTF8, 0, txt, -1, buf, len);
+
+    if (!(out = _wfopen(buf, L"rb"))) {
+        error("Cannot open input file: %s", txt); /* already covered at R level too */ /* #nocov */
+    }
+    return out;
+}
+#else
+FILE* my_open(const char* txt) {
+    FILE* out;
+    if (!(out = fopen(txt, "rb"))) {
```

Comment:
This is the only line that is different.  So how about rewriting this to

```c++
FILE* my_open(const char* txt) {
￼  FILE* out;
#ifdef _WIN32
   wchar_t *buf;
  // ... as a above
   out = _wfopen(buf, L"rb");
#else 
   out = fopen(txt, "rb")
#endif
  if (!out)  {
       error("Cannot open input file: %s", txt);  /* #nocov */
   }
   return out;
}
``` 

which keeps the per-platform code more local?   And come to think about it, I probably take the error check out as well given that we only need it one place (as you righty identified).  

And then, while we're at it, maybe change the not-too-descriptive `my_open` to `open_with_windows_widechar` or something.